pkg/kubelet/pod_workers_test.go

```diff
@@ -640,3 +652,85 @@ func TestKillPodNowFunc(t *testing.T) {
 		t.Errorf("Pod terminated %v, but expected %v", syncPodRecords[1].terminated, true)
 	}
 }
+
+func Test_allowPodStart(t *testing.T) {
+	testCases := []struct {
+		desc                               string
+		pod                                *v1.Pod
+		startedStaticPodsByFullname        map[string]types.UID
+		waitingToStartStaticPodsByFullname map[string][]types.UID
+		allowed                            bool
+	}{
+		{
+			// TODO: Do we want to allow non-static pods with the same full name?
+			// Note that it may disable the force deletion of pods.
+			desc:    "non-static pod",
+			pod:     newPod("test", "test"),
```

Comment:
I expected a copy of this test that verifies if a static and non-static with different uids and the same full name start, both start (one extra test for "static then non-static" and one for "non-static then static" and both should be allowed)

Reply:
I am worried that they will result in a non-static pod and a mirror pod with the same full name in the apiserver so that we may not able to designate one of them - as all kubernetes APIs are based on name / namespace key.

(not sure even it is possible for apiserver(etcd) to allow two pods with the same full name)